src/test/java/mate/academy/spring/boot/controllerTests/BookControllerTest.java

```diff
@@ -0,0 +1,289 @@
+        mockMvc.perform(get("/books/by-title")
+                .param("title", nonExistentTitle))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.length()").value(0));
+    }
+
+    @Test
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @DisplayName("Should create new book, and return BookDto")
+    public void testCreateBook_WithValidRequest_ShouldCreateAndReturnBook() throws Exception {
+        CreateBookRequestDto requestDto = getCreateBookRequestDto("New Book", "Author Name",
+                BigDecimal.valueOf(34), "isbn");
+        BookDto expected = getBookDto("New Book", "Author Name", BigDecimal.valueOf(34));
+        expected.setIsbn("isbn");
+        String jsonRequest = objectMapper.writeValueAsString(requestDto);
+        MvcResult result = mockMvc.perform(post("/books")
+                .content(jsonRequest)
+                .contentType(APPLICATION_JSON))
+                .andExpect(status().isCreated())
+                .andReturn();
+        BookDto actual = objectMapper.readValue(result.getResponse().getContentAsString(), BookDto.class);
+        Assertions.assertNotNull(actual);
+        Assertions.assertNotNull(actual.getId());
+        EqualsBuilder.reflectionEquals(expected, actual, "id");
+    }
+
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @Test
+    @DisplayName("Adding a book with incorrect parameters,should return BadRequest")
+    public void testCreateBook_WhenInvalidRequest_ShouldReturnBadRequest() throws Exception {
+        CreateBookRequestDto invalidRequest = new CreateBookRequestDto();
+        invalidRequest.setTitle("");
+        invalidRequest.setPrice(BigDecimal.valueOf(-1));
+        mockMvc.perform(post("/books")
+                        .contentType(APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(invalidRequest)))
+                .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @DisplayName("Should delete book successfully and return NoContent")
+    public void deleteBook_WhenBookExists_ShouldReturnNoContent() throws Exception {
+        Long bookId = 1L;
+        mockMvc.perform(delete("/books/{id}", bookId))
+                .andExpect(status().isNoContent());
+    }
+
+    @Test
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @DisplayName("Should return 404 Not Found when trying to fetch a deleted book")
+    public void getDeletedBook_WhenBookDoesNotExist_ShouldReturnNotFound() throws Exception {
+        Long bookId = 44L;
+        mockMvc.perform(get("/books/{id}", bookId))
+                .andExpect(status().isNotFound());
+    }
+
+    @Test
+    @WithMockUser
+    @DisplayName("Search for books by the author parameter, we are waiting List of BookDto")
+    public void testSearchBooks_BySearchParameters_ShouldReturnListOfBookDtos() throws Exception {
```

Comment:
Let's split this method using Given-When-Then construction, as liliia-ponomarenko mentioned before